src/tests/cmocka/test_pam_srv.c

```diff
@@ -3186,6 +3186,131 @@ void test_pam_preauth_ocsp_soft_ocsp(void **state)
     assert_int_equal(ret, EOK);
 }
 
+/* With two valid CRL files a certificate should be returned */
```

Comment:
Hi,

the tests are ok, but the comments are wrong.

Here you use 'SSSD_test_CA_crl.pem' and 'SSSD_test_CA_expired_crl.pem', both are from the CA which issued to user certificate but one is expired (invalid) while the other is still valid. As a result the current one is used and the expired one is ignored, validation passes and the certificate is returned.

In the following two tests you are using SSSD_test_CA_expired_crl.pem (expired/invalid) again and SSSD_test_ECC_crl.pem which is valid but from a different CA, so imo it should not ca called invalid. Since the CRL from the CA which issued the user certificate is expired (invalid) the validation fails.

I would like to suggest to add tests with SSSD_test_CA_crl.pem and SSSD_test_ECC_crl.pem, so 2 valid CRLs, as well and check both orders as well to make sure that `X509_load_crl_file()` does not care about the order.  Btw, since the difference is only the in list of CRLs added I would suggest to move the main part of the successful and unsuccessful test into two functions of its own which gets `monitor_params` as input. This should help to avoid code duplication. Additionally, I think  setting `pam_params` is not needed at all.

bye,
Sumit

Reply:
> In the following two tests you are using SSSD_test_CA_expired_crl.pem (expired/invalid) again and SSSD_test_ECC_crl.pem which is valid but from a different CA, so imo it should not ca called invalid. Since the CRL from the CA which issued the user certificate is expired (invalid) the validation fails.

There's a failure in the last part of the first sentence and I'm having problems understanding the complete paragraph: "so imo it should not **ca** called invalid". What do you mean?

Reply:
> > In the following two tests you are using SSSD_test_CA_expired_crl.pem (expired/invalid) again and SSSD_test_ECC_crl.pem which is valid but from a different CA, so imo it should not ca called invalid. Since the CRL from the CA which issued the user certificate is expired (invalid) the validation fails.
> 
> There's a failure in the last part of the first sentence and I'm having problems understanding the complete paragraph: "so imo it should not **ca** called invalid". What do you mean?

ah, sorry, "..., so imo it should not be called invalid."

Reply:
I think I fixed all your concerns. Can you take a look at the tests again and confirm?